src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -113,6 +114,15 @@ func (h *promReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		xhttp.Error(w, rErr.Inner(), rErr.Code())
 		return
 	}
+	ctx = logging.NewContext(ctx,
+		iOpts,
+		zap.String("query", parsedOptions.Params.Query),
```

Comment:
We should decide if we approve of including this in the log context cc @schallert do you have thoughts here? Also we'd want to add this to the other query endpoints for whatever we decide.